model/imageproc/images.go

```diff
@@ -55,6 +58,31 @@ func Resize(img image.Image, newSize image.Point, method int) image.Image {
 	return dst
 }
 
+// Pad returns an image which has been resized to fit within a new size, preserving aspect ratio, and padded with a color.
+func Pad(img image.Image, newSize image.Point, color color.Color, kernel draw.Interpolator) image.Image {
```

Comment:
Does newSize need to be bigger than the original image? It looks like it will work if the image is smaller but I'm not sure if that makes sense?

Reply:
it should work if it's smaller. the image will be scaled accordingly. the main difference between this and resize is this will preserve the aspect ratio while filling in gaps with the specified color while resize will stretch the image